src/hooks/useContract.ts

```diff
@@ -83,6 +110,23 @@ export function useSarStakingContract(): Contract | null {
   return useContract(SAR_STAKING_ADDRESS[chainId], SarStaking, true);
 }
 
+// this is designed for Hedera
+export function useSarNFTStakingContract() {
+  const chainId = useChainId();
+  const sarContractAddress = SAR_STAKING_ADDRESS[chainId];
+
+  let address: string | undefined = undefined;
+  if (sarContractAddress) {
+    // for hedera we need to add 1 to contract address to get the nft/token contract address, it happens because the nft/tokens have different contract
+    // eg. 0.0.49031363 is contract address it's only possible to call positions, rewardRate, etc...
+    // 0.0.49031364 is nft contract address it's only possible to call tokenURI, balanceOf, etc..., the available fuctions of ERC721 interface
+    const _address = (Number(sarContractAddress) + 1).toString(16);
+    // need to fill rest of address to get valid address
+    address = getAddress('0x' + new Array(41 - _address.length).join('0') + _address);
+  }
+  return useHederaContract(address, SarStaking, true);
```

Comment:
in `useSarNFTStakingContract`, you need to reuse `contractToTokenId` method from `src/utils/hedera.ts` utility. Also you don't need to use `toString(16)`, we already have function `hederaId` in `src/utils/hedera.ts` which takes solidity address and return hedera id format, so it will look like this
```ts
const sarContractId = hedera.hederaId(sarContractAddress)
const nftTokenId = hedera.contractToTokenId(sarContractId)
const nftTokenAddress = hedera.idToAddress(nftTokenId)

return useContract(nftTokenAddress, SarStaking);
```